lib/src/widgets/type_indicator_widget.dart

```diff
@@ -0,0 +1,247 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+
+import 'package:chat_ui/src/widgets/profile_circle.dart';
+
+import '../../chat_ui.dart';
+
+class TypingIndicator extends StatefulWidget {
+  const TypingIndicator({
+    Key? key,
+    this.showIndicator = false,
+    this.profilePic,
+    this.chatBubbleConfig,
+    this.typeIndicatorConfig,
+  }) : super(key: key);
+
+  final bool showIndicator;
+  final String? profilePic;
+  final ChatBubble? chatBubbleConfig;
+  final TypeIndicatorConfiguration? typeIndicatorConfig;
+
+  @override
+  _TypingIndicatorState createState() => _TypingIndicatorState();
+}
+
+class _TypingIndicatorState extends State<TypingIndicator>
+    with TickerProviderStateMixin {
+  late AnimationController _appearanceController;
+
+  late Animation<double> _indicatorSpaceAnimation;
+
+  late Animation<double> _largeBubbleAnimation;
+
+  late AnimationController _repeatingController;
+  final List<Interval> _dotIntervals = const [
+    Interval(0.25, 0.8),
+    Interval(0.35, 0.9),
+    Interval(0.45, 1.0),
+  ];
+
+  final List<AnimationController> _jumpControllers = [];
```

Comment:
dispose

Reply:
This list of animationcontroller, we need to find solution for this